src/server/worker/pipeline/transform/transform_test.go

```diff
@@ -219,14 +208,25 @@ func triggerJob(t *testing.T, env *testEnv, pi *pps.PipelineInfo, files []taruti
 	}))
 	require.NoError(t, env.PachClient.PutFileTAR(commit, buf, client.WithAppendPutFile()))
 	require.NoError(t, env.PachClient.FinishCommit(pi.Details.Input.Pfs.Repo, commit.Branch.Name, commit.ID))
+	return commit
+}
+
+func deleteFiles(t *testing.T, env *testEnv, pi *pps.PipelineInfo, files []string) *pfs.Commit {
+	commit, err := env.PachClient.StartCommit(pi.Details.Input.Pfs.Repo, "master")
+	require.NoError(t, err)
+	for _, file := range files {
+		require.NoError(t, env.PachClient.DeleteFile(commit, file))
+	}
+	require.NoError(t, env.PachClient.FinishCommit(pi.Details.Input.Pfs.Repo, commit.Branch.Name, commit.ID))
+	return commit
 }
 
 func testJobSuccess(t *testing.T, env *testEnv, pi *pps.PipelineInfo, files []tarutil.File) {
-	ctx, jobInfo := mockBasicJob(t, env, pi)
-	triggerJob(t, env, pi, files)
+	commit := writeFiles(t, env, pi, files)
+	ctx, jobInfo := mockJobFromCommit(t, env, pi, commit)
 	ctx = withTimeout(ctx, 10*time.Second)
 	<-ctx.Done()
-	require.Equal(t, pps.JobState_JOB_SUCCESS, jobInfo.State)
+	require.Equal(t, pps.JobState_JOB_FINISHING, jobInfo.State)
```

Comment:
maybe a comment about why finishing instead of success (we're only testing the worker code here, success is marked by pfs master), could go here or at the top of the tests